File: app/src/main/java/org/uadaf/app/internal/view/fabs/FAB.java
package org.uadaf.app.internal.view.fabs;

import android.content.Context;
import android.graphics.Matrix;
import android.util.AttributeSet;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class FAB extends FloatingActionButton {

    private Matrix imageMatrix;

    public FAB(Context context) {
        super(context);
    }

    public FAB(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public FAB(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();
        imageMatrix = getImageMatrix();
    }

    @Override
    public void setImageResource(int resId) {
        super.setImageResource(resId);
        setImageMatrix(imageMatrix);
    }
}

File: app/src/main/java/org/uadaf/app/internal/view/fabs/FAB.kt
package org.uadaf.app.internal.view.fabs

import android.content.Context
import android.graphics.Matrix
import android.util.AttributeSet
import com.google.android.material.floatingactionbutton.FloatingActionButton

open class FAB: FloatingActionButton {

    private var mMatrix: Matrix? = null

    constructor(context: Context): super(context)

    constructor(context: Context, attrs: AttributeSet): super(context, attrs)

    constructor(context: Context, attrs: AttributeSet, defStyles: Int): super(context, attrs, defStyles)

    override fun onFinishInflate() {
        super.onFinishInflate()
        mMatrix = imageMatrix
    }

    override fun setImageResource(resId: Int) {
        super.setImageResource(resId)
        imageMatrix = mMatrix
    }

}